packages/core/echo/echo-pipeline/src/db-host/doc-sync-state.ts

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { next as A } from '@dxos/automerge/automerge';
+import { type DocHandle, type Heads } from '@dxos/automerge/automerge-repo';
+
+/**
+ * Class to synchronize docs between host and thin client.
+ */
+export class DocSyncState<T> {
```

Comment:
Does this need to be a separate class -- rather than just state managed by the synchronizer?